src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -3651,7 +3657,7 @@ public void ListPattern_StringLength()
 class C
 {
     public string Length => throw null;
-    public int this[int i] => throw null;
+    public int this[System.Index i] => throw null;
```

Comment:
>System.Index [](http://example.com/codeflow?start=20&length=12)

I think we want to keep the original test as well. Consider adding a new unit test instead of changing the scenario.